hltDiff: compare TriggerResults event by event (76x)

fwyzard opened the pull request:
Compare two TriggerResults collections event by event.
These can come from two collections in the same file(s), or from two different sets of files.

fwyzard committed:
hltDiff: compare TriggerResults event by event

Compare two TriggerResults collections event by event.
These can come from two collections in the same file(s), or from two
different sets of files.

cmsbuild commented:
A new Pull Request was created by @fwyzard (Andrea Bocci) for CMSSW_7_6_X.

hltDiff: compare TriggerResults event by event (76x)

It involves the following packages:

HLTrigger/Tools


@Martin-Grunewald, @perrotta, @cmsbuild, @fwyzard can you please review it and eventually sign? Thanks.
@Martin-Grunewald this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

fwyzard committed:
hltDiff: add GNU-style long options

fwyzard committed:
hltDiff: use HLTConfigData to access the trigger information

fwyzard committed:
hltDiff: add special handling of HLTPrescaler modules

fwyzard committed:
hltDiff: save memory by allocating module types in a set, and storing pointers to them in the vectors

fwyzard committed:
hltDiff: identify old/new collections by process name

fwyzard committed:
hltDiff: add a one-line summary with the number of affected events

Martin-Grunewald commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #11510 was updated. @Martin-Grunewald, @perrotta, @cmsbuild, @fwyzard can you please check and sign again.

cmsbuild commented:
+1
Tested at: 9456df9632d0c2eeb2eaff7cbbf038f5a66b7a67
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11510/8453/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11510/8453/summary.html

fwyzard committed:
hltDiff: implement partial comparison for different HLT menus

In case the "old" and "new" HLT menus are different, only the common
paths are compared.

fwyzard committed:
hltDiff: print trigger candidates for filters with a different decision

fwyzard committed:
hltDiff: print trigger candidates for events with a different decision

fwyzard commented:
+1

fwyzard commented:
please test

fwyzard commented:
though this only adds a new stand-alone command line tool, so there is nothing to be tested...

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_6_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

cmsbuild commented:
Pull request #11510 was updated. @cmsbuild can you please check and sign again.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_6_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

cmsbuild commented:
+1
Tested at: 9358824478ede527e17fa5b183c407b8058b18f6
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11510/8530/summary.html

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_6_X IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11510/8530/summary.html

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_6_X IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

davidlange6 commented:
+1
